src/lib/serviceRoutines/postQueryContext.cpp

```diff
@@ -49,115 +52,57 @@ static char* xmlPayloadClean(const char*  payload, const char* payloadWord)
-
-  ciP->httpStatusCode = mongoQueryContext(&parseDataP->qcr.res, qcrP, ciP->tenant, ciP->servicePathV, ciP->uriParam);
-
-  // If no redirectioning is necessary, just return the result
-  if (qcrP->errorCode.code != SccFound)
-  {
-    //
-    // In case the response is empty, fill response with the entity we looked for
-    //
-    if (qcrP->contextElementResponseVector.size() == 0)
-    {
-      //
-      // FIXME P6: What do we do if we have more than one entity in incoming entityIdVector?
-      //           Push each entityId in a separate contextElementResponse?
-      //           Also, perhaps mongoQueryContext should implement this and not postQueryContext
-      //           This FIXME is related to github issue #588 and (probably) #650.
-      //           Also, optimizing this would be part of issue #768
-      //
-
-
-      //
-      // It's better not to give any details here as we'd have to choose ONE of the entities in the vector ...
-      // std::string details = "Entity id: /" + parseDataP->qcr.res.entityIdVector[0]->id + "/";
-      //
-      qcrP->errorCode.fill(SccContextElementNotFound);
-    }
-
-    answer = qcrP->render(ciP, QueryContext, "");
-    return answer;
-  }
-
-
-
   std::string     ip;
   std::string     protocol;
   int             port;
   std::string     prefix;
-
-  //
-  // An entity/attribute has been found on some context provider.
-  // We need to forward the query request to the context provider, indicated in qcrP->errorCode.details.
-  //
-  // 1. Parse the providing application to extract IP, port and URI-path
-  // 2. The exact same QueryContextRequest will be used for the query-forwarding
-  // 3. Render an XML-string of the request we want to forward
-  // 4. Send the request to the providing application (and await the response)
-  // 5. Parse the XML response and fill in a binary QueryContextResponse
-  // 6. Render QueryContextResponse from the providing application
-  //
+  std::string     answer;
 
 
   //
   // 1. Parse the providing application to extract IP, port and URI-path
   //
-  if (parseUrl(qcrP->errorCode.details, ip, port, prefix, protocol) == false)
+  if (parseUrl(qcrP->contextProvider, ip, port, prefix, protocol) == false)
   {
-    QueryContextResponse qcrs;
-
-    LM_W(("Bad Input (invalid proving application '%s')", qcrP->errorCode.details.c_str()));
+    LM_W(("Bad Input (invalid providing application '%s')", qcrP->contextProvider.c_str()));
     //  Somehow, if we accepted this providing application, it is the brokers fault ...
     //  SccBadRequest should have been returned before, when it was registered!
 
-    qcrP->errorCode.fill(SccContextElementNotFound, "");
-    answer = qcrP->render(ciP, QueryContext, "");
-    return answer;
+    qcrsP->errorCode.fill(SccContextElementNotFound, "");
+    return;
   }
 
 
   //
-  // 2. The exact same QueryContextRequest will be used for the query-forwarding
+  // 2. Render an XML-string of the request we want to forward
```

Comment:
I understand this is now hardwired to XML as this is the only format support right now, but it would be more flexible when also JSON becaomes a possibility (i.e. fixing https://github.com/telefonicaid/fiware-orion/issues/722), right?